rasa/core/actions/action.py

```diff
@@ -95,17 +97,26 @@ def action_for_index(
         The instantiated `Action` or `None` if no `Action` was found for the given
         index.
     """
+    index = prediction.max_confidence_index
     if domain.num_actions <= index or index < 0:
         raise IndexError(
             f"Cannot access action at index {index}. "
             f"Domain has {domain.num_actions} actions."
         )
 
-    return action_for_name(domain.action_names[index], domain, action_endpoint)
+    return action_for_name(
+        domain.action_names[index],
+        domain,
+        action_endpoint,
+        prediction.is_end_to_end_prediction,
+    )
 
 
 def action_for_name(
-    action_name: Text, domain: Domain, action_endpoint: Optional[EndpointConfig]
+    action_name: Text,
+    domain: Domain,
+    action_endpoint: Optional[EndpointConfig],
+    is_end_to_end_prediction: bool = False,
```

Comment:
in `Prediction` class `is_end_to_end_prediction` means that a prediction was made using user text, is it the same here?
In this case it shouldn't affect whether it is `action_text` or `action_name`, It's domain's job to differentiate based on given index